c2rust-ast-exporter/src/lib.rs

```diff
@@ -95,6 +95,7 @@ extern "C" {
     fn clang_version() -> *const c_char;
 }
 
+#[allow(clippy::unnecessary_cast)]
```

Comment:
I believe you can put this on the specific line that needs it.  Also, can you add a `reason = ""`, but commented out with `/* , reason = "" */` since it's not stabilized yet (but will be soon after merging #1227)?  Something like "needed on aarch64-apple-darwin" seems good.

Reply:
Also, if `cfg_attr` works for this to only enable it when needed, even better.  Not sure that works, but it might.